screens/main/home.js

```diff
@@ -35,67 +35,60 @@ export default function HomeScreen({navigation}) {
   const [recordedWalks, setRecordedWalks] = useState(null);
   const [activeWalk, setActiveWalk] = useState(null);
 
-  const saveStepsAndDistances = () => {
-    Realm.getContest().then(newContest => {
-      const today = moment().endOf('day');
-      let from = null,
-        to = null;
-      if (newContest) {
-        /// check if we're in/after the contest period
-        if (!newContest.isBeforeStartDate) {
-          from = moment(newContest.start);
-          /// check if we're in the contest period
-          if (newContest.isAfterEndDate) {
-            to = moment(newContest.end);
-          } else {
-            to = today;
-          }
+  async function saveStepsAndDistances() {
+    const newContest = await Realm.getContest();
+    const today = moment().endOf('day');
+    if (newContest) {
+      let from = null;
+      let to = null;
+      /// check if we're in/after the contest period
+      if (!newContest.isBeforeStartDate) {
```

Comment:
@emdat basically all that I think that will need to happen is updating this "start date" check to instead check the baseline collection start date, and to collect/store/upload from baseline collection start date up to the current date (or contest end, whichever is earlier).